File: Benchmark/Go/Compute/FibonacciLoop.go
package main

import (
 "fmt"
)

// Fibonacci calculates the nth Fibonacci number using an iterative approach
func fibonacci(n int) int64 {
 if n <= 1 {
  return int64(n)
 }

 var a, b int64 = 0, 1

 for i := 2; i <= n; i++ {
  temp := a + b
  a = b
  b = temp
 }

 return b
}

func main() {
 // Example usage
 n := 10
 fmt.Printf("The %dth Fibonacci number is: %d\n", n, fibonacci(n))

 // Print first 20 Fibonacci numbers
 fmt.Println("First 20 Fibonacci numbers:")
 for i := 0; i < 20; i++ {
  fmt.Printf("%d ", fibonacci(i))
 }
 fmt.Println()
}

File: Benchmark/Rust/Compute/FibonacciLoop.rs
fn fibonacci(n: u32) -> u64 {
    if n <= 1 {
        return n as u64;
    }

    let mut a = 0;
    let mut b = 1;

    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }

    b
}

fn main() {
    // Example usage
    let n = 10;
    println!("The {}th Fibonacci number is: {}", n, fibonacci(n));

    // Print first 20 Fibonacci numbers
    println!("First 20 Fibonacci numbers:");
    for i in 0..20 {
        print!("{} ", fibonacci(i));
    }
    println!();
}
